### PR TITLE
Add `read_args` and `store_args` to `cache_args`

### LINKED ISSUE #110: ENH: Add `read_args` and `store_args` to `cache_args`
Libraries like pandas and polars usually have some configuration options for how to read and write data. We should make sure that users still have access to those options when using our cacher.

### PR DESCRIPTION
**Pull request by Jeffrey-Lim**
Closes #110. Allows the user to supply read and store arguments with the cacher.

### CODE REVIEW

**Comment by schobbejak** [comment]
I'm going to move this to v0.3 for now since it seems like it was not high priority

**Comment by tolgakopar** [comment]
Pipeline fails, not up to date with branch v0.3 and why change the class name, there's no point

File: epochalyst/_core/_caching/_cacher.py [hunk 1]
```diff
@@ -11,8 +12,13 @@
...
 from epochalyst._core._logging._logger import _Logger
 
+if sys.version_info < (3, 11):
+    from typing_extensions import NotRequired
+else:
+    from typing import NotRequired
 
-class _CacheArgs(TypedDict):
+
+class CacheArgs(TypedDict):
```
[the remark this answers is not in the document]
Comment (schobbejak) [reply epochalyst/_core/_caching/_cacher.py:36]: Alright once I configured pre-commit it started complaining

**Review by schobbejak — APPROVED** [review]
Looks fine

**Pull request merged**